core/src/main/scala/pureconfig/DurationConvert.scala

```diff
@@ -42,9 +43,11 @@ private[pureconfig] object DurationConvert {
    */
   def fromDuration(d: Duration): String = {
     d match {
+      case f: FiniteDuration => fromFiniteDuration(f)
+      // We must do an `eq` instead of `==` comparison because Undefined is intentionally != itself.
+      case i: Duration.Infinite if i eq Duration.Undefined => Duration.Undefined.toString
```

Comment:
@ruippeixotog Excellent point. Thank you. Done.
I had been wanted to do something better, but wasn't seeing the solution.